webapp/lib/view.dart

```diff
@@ -227,6 +228,77 @@ class ProjectSelectorView {
   }
 }
 
+class ChartFiltersView {
+  static final ChartFiltersView _singleton = new ChartFiltersView._internal();
+
+  factory ChartFiltersView() {
+    return _singleton;
+  }
+
+  DivElement chartFiltersContainer;
+  DivElement _singleFilterSpan;
+  LabelElement _periodFilterTitle;
+  SelectElement _periodFilter;
+
+  ChartFiltersView._internal() {
+    chartFiltersContainer = new DivElement()..classes.add('chart-filters');
+    _singleFilterSpan = new DivElement()..classes.add('chart-filter');
+    _periodFilterTitle = new LabelElement()..text = 'Filter by Period:';
+    _periodFilter = new SelectElement()..classes.add('period-filter');
+    _periodFilter.children.addAll(_getPeriodFilterOptions());
+    _periodFilter.onChange.listen(filterChartsByPeriod);
```

Comment:
I'm guessing this is why you added an argument to `filterChartsByPeriod`? You can rewrite this as:
`(_) => filterChartsByPeriod()` which is still very clear and tidy.

Reply:
fixed